op-devstack/dsl/l2_network.go

```diff
@@ -67,31 +67,7 @@ func (n *L2Network) CatchUpTo(o *L2Network) {
 }
 
 func (n *L2Network) WaitForBlock() {
-	l2_el := n.inner.L2ELNode(match.FirstL2EL)
-
-	initial, err := l2_el.EthClient().InfoByLabel(n.ctx, eth.Unsafe)
-	n.require.NoError(err, "Expected to get latest block from L2 execution client")
-
-	initialHash := initial.Hash()
-
-	err = wait.For(n.ctx, 1000*time.Millisecond, func() (bool, error) {
-		latest, err := l2_el.EthClient().InfoByLabel(n.ctx, eth.Unsafe)
-		if err != nil {
-			return false, err
-		}
-
-		newHash := latest.Hash()
-
-		if initialHash.Cmp(newHash) == 0 {
-			n.log.Info("Still same block detected", "number", latest.NumberU64(), "chain", n.ChainID(), "initial_block_hash", initialHash, "new_block_hash", newHash)
-
-			return false, nil
-		}
-
-		n.log.Info("New block detected", "chain", n.ChainID(), "prev_block_hash", initialHash, "new_block_hash", newHash, "time", latest.Time())
-		return true, nil
-	})
-	n.require.NoError(err, "Expected to get latest block from L2 execution client for comparison")
+	NewL2ELNode(n.inner.L2ELNode(match.FirstL2EL)).WaitForBlock()
```

Comment:
nit: Is it desirable/possible to setup a reference to this dsl.L2ELNode somewhere on L2Network so it doesn't need to be recreated every time?

Reply:
centralizing discussion [here](https://github.com/ethereum-optimism/optimism/pull/15914#discussion_r2089191228)